crates/rpc/src/handlers/validator.rs

```diff
@@ -375,3 +375,75 @@ pub async fn post_validator_balances_from_state(
         ))),
     )
 }
+
+#[derive(Debug, Serialize)]
+pub struct ValidatorLivenessData {  
+    #[serde(with = "serde_utils::quoted_u64")]  
+    pub index: u64,  
+    pub is_live: bool,  
+}
+
+impl ValidatorLivenessData {  
+    pub fn new(index: u64, is_live: bool) -> Self {  
+        Self { index, is_live }  
+    }  
+}  
+
+#[post("/validator/liveness/{epoch}")]    
+pub async fn post_validator_liveness(    
+    db: Data<ReamDB>,    
+    epoch: Path<u64>,    
+    validator_indices: Json<Vec<String>>,    
+) -> Result<impl Responder, ApiError> {    
+    let epoch = epoch.into_inner();    
+    let validator_indices = validator_indices.into_inner();    
+        
+    let slot = epoch * SLOTS_PER_EPOCH;    
+    let state = get_state_from_id(ID::Slot(slot), &db).await?;    
+        
+    let mut liveness_data = Vec::new();    
+        
+    for validator_index_str in validator_indices {    
+        let validator_index: u64 = validator_index_str.parse()  
+            .map_err(|_| ApiError::BadRequest("Invalid validator index".to_string()))?;  
```

Comment:
instead of parsing like this could we do

```
struct ValidatorIndex(pub #[serde(with = "serde_utils::quoted_u64")]  u64);

validator_indices: Json<Vec<ValidatorIndex>>
```


just so we can avoid invalid inputs earlier on, no point in doing work if the input is invalid from the start